src/sentry/taskworker/retry.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from enum import Enum
+from multiprocessing.context import TimeoutError
+
+from sentry_protos.sentry.v1.taskworker_pb2 import RetryState
+
+
+class RetryError(Exception):
+    """
+    Exception that tasks can raise to indicate that the current task
+    should be retried.
+    """
+
+
+class FinalAction(Enum):
```

Comment:
I'm not a fan of this name either, but I also don't have a good alternative. Some other not so great options:
* `times_exceeded_action`? 
* If `Retry.times` -> `Retry.quota`, then `quota_exceeded_action`?

Reply:
Yeah `final_action` isn't a great name. I like 'times_exceeded' though.